charms/jupyter-ui/src/charm.py

```diff
@@ -210,7 +261,7 @@ def _on_pebble_ready(self, _):
             return
 
         # upload files to container
-        self._upload_files_to_container()
```

Comment:
There a reason why we'd upload the logos on pebble_ready, but not the image lists?

Reply:
the logos is pushed on pebble ready because it's a static file that is expected to only change on upgrade, while the image lists needs to be refreshed on config changed, on pebble ready, on install etc so it is called on every charm reconciliation (main)

Reply:
ok i see what you mean.  But what happens if we have the sequence:

* install
* config-changed
* pebble-ready

what image list will we have in the UI?

Reply:
Oh wait I think its ok, because _on_pebble_ready calls main as well.  If you agree, then you can resolve this comment

Reply:
It feels a bit like the management of the pebble container is split into a few different places rather than one, so its a little hard to understand what is happening to the container when reading code.  That's not something you've caused, I think you've more inherited this problem from the existing code setup.  And you make a good point that config changes will cause the list to change

What you have now I think makes sense.  with more time, I'd suggest attempting a refactor of the logic, but not needed here